samples/CognitiveSearch/Generated/DataSourcesClient.cs

```diff
@@ -44,14 +44,15 @@ internal DataSourcesClient(ClientDiagnostics clientDiagnostics, HttpPipeline pip
         /// <param name="dataSource"> The definition of the datasource to create or update. </param>
         /// <param name="requestOptions"> Parameter group. </param>
         /// <param name="accessCondition"> Parameter group. </param>
+        /// <param name="matchConditions"> The content to send as the request conditions of the request. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual async Task<Response<DataSource>> CreateOrUpdateAsync(string dataSourceName, Enum0 prefer, DataSource dataSource, Models.RequestOptions requestOptions = null, AccessCondition accessCondition = null, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<DataSource>> CreateOrUpdateAsync(string dataSourceName, Enum0 prefer, DataSource dataSource, Models.RequestOptions requestOptions = null, AccessCondition accessCondition = null, MatchConditions matchConditions = null, CancellationToken cancellationToken = default)
```

Comment:
Why was this matchConditions parameter added when there wasn't a conditional request header in the method signature before?